lib/erb.rb

```diff
@@ -548,7 +549,7 @@ class Buffer # :nodoc:
       def initialize(compiler, enc=nil)
         @compiler = compiler
         @line = []
-        @script = enc ? "#coding:#{enc}\n" : ""
+        @script = enc ? "#coding:#{enc}\n".dup : String.new
```

Comment:
Now, I'm travelling so I can't take a look at on my mac. But what I remember, the instance variable is modified by `<<` in the other methods such as `#close` and `#cr`. So, the `dup` is needed if an encoding is set. Please, correct me if I'm wrong.